test/unit/helpers/multi.spec.ts

```diff
@@ -0,0 +1,105 @@
+import * as path from "path";
+import * as util from "../../util";
+import * as tstl from "../../../src";
+import {
+    invalidMultiFunctionUse,
+    invalidMultiTypeToNonArrayBindingPattern,
+    invalidMultiTypeArrayBindingPatternElementInitializer,
+} from "../../../src/transformation/utils/diagnostics";
+
+const multiProjectOptions: tstl.CompilerOptions = {
+    types: [path.resolve(__dirname, "../../../helpers")],
+};
+
+test.each<[string, any]>([
+    ["$multi()", undefined],
+    ["$multi(true)", true],
+    ["$multi(1, 2)", 1],
+])("$multi call on return statement (%s)", (expression, result) => {
+    util.testFunction`
+        return ${expression};
+    `
+        .setOptions(multiProjectOptions)
+        .expectToEqual(result);
+});
+
+const multiFunction = `
+function multi(...args) {
+    return $multi(...args);
+}
+`;
+
+const createCasesThatCall = (name: string): Array<[string, any]> => [
+    [`let a; [a] = ${name}()`, undefined],
+    [`const [a] = ${name}()`, undefined],
+    [`const [a] = ${name}(1)`, 1],
+    [`const ar = [1]; const [a] = ${name}(...ar)`, 1],
+    [`const _ = null, [a] = ${name}(1)`, 1],
+    [`let a; for (const [a] = ${name}(1, 2); false; 1) {}`, undefined],
+    [`let a; for ([a] = ${name}(1, 2); false; 1) {}`, 1],
+];
+
+test.each<[string, any]>(createCasesThatCall("$multi"))("invalid direct $multi function use (%s)", statement => {
+    util.testModule`
+        ${multiFunction}
+        ${statement}
+        export { a };
+    `
+        .setOptions(multiProjectOptions)
+        .setReturnExport("a")
+        .expectDiagnosticsToMatchSnapshot([invalidMultiFunctionUse.code]);
+});
```

Comment:
Personally I think these tests are really hard to read/understand. But that might be because I haven't done a lot of js/ts lately.